app-creator/data/templates/templates_test.go

```diff
@@ -0,0 +1,149 @@
+package templates_data
+
+import (
+	"testing"
+	"net/http/httptest"
+	"io/ioutil"
+	"net/http"
+	"encoding/json"
+	"strings"
+)
+
+func TestToJSON(t *testing.T) {
```

Comment:
I would make this and test name below more descriptive: 

`TestTemplateSerializationToJSON`

`TestTemplateDeserializationFromJSON`